### ISSUE TITLE
Names of relocations used by the overlay system

### ISSUE THREAD

**Issue by edward-jones** [riscv-software-src/riscv-overlay#29]
Currently the relocations have the following names:

```
R_RISCV_OVL_HI20
R_RISCV_OVL_LO12_I
R_RISCV_OVL32
R_RISCV_OVLPLT_HI20
R_RISCV_OVLPLT_LO12_I
R_RISCV_OVLPLT32
```

I propose that it might be better to rename the `R_RISCV_OVL_*` relocations to be `R_RISCV_OVLTOK_*`. My motive for this is that by using 'OVLTOK' it's a bit clearer that the value being encoded is a token value rather than an address. It also draws a clearer distinction from the `OVLPLT` relocations which really *are* addresses to a resident thunk.

**Comment by oferShinaar** [comment]
Make sense. 
I am assigning this to @Craig. We will need to update the HLD as well.
Thanks, Ed

**Comment by edward-jones** [comment]
Another question here. Should we remove the 32 from the names of the relocations. If tokens can *only* be 32-bits then maybe this is unnecessary since it implies the potential existence of a 64-bit relocation?

**Comment by edward-jones** [comment]
I've opened a merge request to do this #31

**Comment by oferShinaar** [comment]
As you mentioned, the 32bit postfix is for highlighting the length of the token; it does not imply it is for 32 machines. The length of the token is a derivative of how many `groups` we can have in the program and also can give a wider `offset` to reach far functions in the group.
If we will want to have 48 or 64bit token in the future, it is possible since it is not restricted on the [overlay-software-standard.adoc](https://github.com/fossi-foundation/embedded-sw-overlay/blob/master/docs/overlay-software-standard.adoc)

Bottom line I agree, we should remove the 32 postfixes just because we can have different sizes of tokens in the future, and those will be encapsulated to the same relocation ID

**Comment by edward-jones** [comment]
If later it was possible to have different sized tokens then I assume this would be decoupled from the XLEN? If that's the case how would you ensure ABI compatibility across different object files which may disagree about the size of the relocation? That could be achieved by using an attribute or a different ABI

The tools are going to need to know how wide the underlying token is anyway, since they need to generate a load to it. I'm not sure there's a particular need to try and minimize the number of relocations used either.

**Comment by oferShinaar** [comment]
As explained, the 32bit postfix is for the token size. It is not related to the machine architecture; therefore, it should not be decoupled with XLEN.
The current overlay implementation (per HLD) can only work on 32bit [machines](https://github.com/riscv-software-src/riscv-overlay/blob/master/docs/overlay-hld.adoc#26-rt-engine).

To support token size other than 32bit, we need more changes in the toolchain + engine. Currently, the HLD is only addressing 32bit tokens. 
This means that tools should follow the HLD and only support 32bit tokens.

The generic overlay standard is not inforcing token size and/or implementation. Therefore other targets or other implementations can be with different token size
